stonesoup/models/measurement/nonlinear.py

```diff
@@ -1248,3 +1250,294 @@ def pdf(self, state1, state2, **kwargs):
+        pred_meas = self.function(state2, **kwargs)
+        if state1.state_vector[0] <= self.sensing_threshold:
+            pdf = p_m + ((1-p_m) * 1/2 * (1+erf((self.sensing_threshold - pred_meas)
+                                                / (nd_sigma * sqrt(2)))))
+            likelihood = np.atleast_1d(np.log(pdf)).view(np.ndarray)
+
+        else:
+            d_sigma = self.standard_deviation_percentage * pred_meas + self.min_noise
+            with np.errstate(divide="ignore"):
+                likelihood = np.atleast_1d(np.log(1/(d_sigma*np.sqrt(2*np.pi)) *
+                                                  np.exp((-(state1.state_vector-pred_meas)
+                                                         ** 2)/(2*d_sigma**2)))).view(np.ndarray)
+
+        if len(likelihood) == 1:
+            likelihood = likelihood[0]
+
+        return likelihood
+
+    def pdf(self, state1: State, state2: State, **kwargs) -> Union[Probability, np.ndarray]:
+        r"""Model pdf/likelihood evaluation function
+
+        Evaluates the pdf/likelihood of ``state1``, given the state
+        ``state2`` which is passed to :meth:`function()`.
+
+        This function implements the following likelihood function,
+        adapted from (12) in [2]_, removing the background sensor noise
+        term.
+
+        .. math::
+            p(z_k|\Theta_k) = \frac{1}{\sigma_d\sqrt{2\pi}}\text{exp}
+            \left(-\frac{(z_k-\hat{z}_k)^2}{2\sigma_d}\right),
+
+        where :math:`z_k` = ``state1``, :math:`\Theta_k` = ``state2``,
+        :math:`\hat{z}_k` = :meth:`~.Model.function` on ``state2``
+        and :math:`\sigma_d` is the measurement standard deviation
+        assuming the measurement arose from a true gas detection.
+        This is given by
+
+        .. math::
+            \sigma_d = \sigma_{\text{percentage}} \cdot \hat{z} + \nu_{\text{min}},
+
+        where :math:`\sigma_{\text{percentage}}` = :attr:`standard_deviation_percentage`
+        and :math:`\nu_{\text{min}}` = :attr:`noise`. In the
+        event that a measurement is below the sensor threshold or
+        missed, a different liklihood function is used. This is
+        given by
+
+        .. math::
+            p(z_k|\Theta_k) = (P_m) + \left((1-P_m)\cdot\frac{1}{2}\left[1+\text{erf}
+            \left(\frac{z_{\text{thr}}-\hat{z}_k}{\sigma_m\sqrt{2}}\right)\right]\right),
+
+        where :math:`P_m` = :attr:`missed_detection_probability`,
+        :math:`\sigma_m` is the missed detection standard deviation
+        which is implemented as equal to :attr:`sensing_threshold`
+        and :math:`\text{erf}()` is the error function.
+
+        Parameters
+        ----------
+        state1 : :class:`~.State`
+        state2 : :class:`~.State`
+
+        Returns
+        -------
+        : :class:`~.Probability`
+            The likelihood of ``state1``, given ``state2``
+        """
+        return Probability.from_log_ufunc(self.logpdf(state1, state2, **kwargs))
```

Comment:
Just to make it clear that this is doing same as parent, could use this:
```suggestion
        return super().pdf(state1, state2, **kwargs)
```